rfcs/20201014-VTune-ITT-tagging/README.md

```diff
@@ -0,0 +1,139 @@
+**Compilation Time**  
+
+We enable this feature on compilation time by default.  
+User can disable the feature by following CMake declaration at compiling time.    
+DNNL_ENABLE_ITT_TASKS=False 
+
+**Runtime**  
+
+Users can use a environment variable to control ITT level at runtime.  
+
+| DNNL_ITT_TASK_LEVEL   | __itt_task_level  | Note |  
+|---------------|---------------------|---------------------|  
+| 0   | __itt_task_level_none |   no ITT event will be triggered  | 
+| 1 | __itt_task_level_low  |    ITT events are only triggered in master thread     | 
+| 2 | __itt_task_level_high |    ITT events are triggered in all OMP/TBB threads      | 
+
+
+### Testing aspects
+---
+**Performance Number**  
+
+
+We average 20 runs for each number.
+For the last two tests, we made the tests without/with VTune running.
+Due to profiling size limitation, VTune profiling ended very early, so we regarded the number is not applicable.
+
+
+| Primitive     | Benchdnn input file                            | Baseline (no patch) | Patch + no VTune + DNNL_ITT_TASK_LEVEL=2| Patch + Vtune + DNNL_ITT_TASK_LEVEL=2 |
+|---------------|------------------------------------------------|---------------------|------------------|---------------|
+| Convolution   | tests/benchdnn/inputs/conv/set_perf_cpu_all_mb | 2858.6515 (stddev : 88.3)    | 2785.1985 ( stddev: 83.13)      | NA (excess 1G data size very quickly)   |
+| Inner Product | tests/benchdnn/inputs/ip/perf_ip_inference_sb  | 349.4827 (stddev : 41.64)         | 349.16535 (stddev : 49.15)   | NA (excess 1G data size very quickly) |
+
+
+For performance numbers with VTune running, we used much smaller tests to get VTune profiling results from start to end.
+
+| Primitive     | Benchdnn input file                            | Baseline (no patch) | Patch + no VTune + DNNL_ITT_TASK_LEVEL=0| Patch + no VTune + DNNL_ITT_TASK_LEVEL=2| Patch + Vtune + ITT_TASK_LEVEL=2 |  
+|---------------|------------------------------------------------|---------------------|------------------|------------------|---------------|
+| convolution | tests/benchdnn/inputs/conv/shapes_alexnet        | 79.93 ms | 80.76 ms| 79.34 ms | 88.25 ms | 
+| ip | tests/benchdnn/inputs/ip/shapes_alexnet        | 4.45 ms | 4.598 ms |4.57 ms |  6.002 ms |
+| bnorm | tests/benchdnn/inputs/bnrom/shapes_resnet_50       | 51.22 ms | 51.25 ms|50.82 ms |  68.19 ms |
+| matmul | tests/benchdnn/inputs/matmul/shapes_3d       | 74.7 ms | 81 ms |77.1 ms |  81.17 ms |
+| softmax | tests/benchdnn/inputs/softmax/shapes_2d       | 125.29 ms | 127.9 ms|126.48 ms |  145.96 ms |
+  
+We indeed saw overhead when we have VTune running, and the slow down is around 75% - 90%.  
+However, the performance with/without this feature is almost the same when we don't have VTune running.  
+**In short, we don't see overhead to enable the feature by default without VTune running.**  
+
+
+## Open Questions  
+
+1. Should we set DNNL_ITT_TASK_LEVEL as __itt_task_level_high by default?
```

Comment:
I think using `__itt_task_level_high` as default would be useful, however feedback from VTune team would be very welcome here.

Reply:
The feedback from VTune team : "Looked into the result – it contains ~110K tasks which is not that much for 15-seconds collection on big server machine (112 cores). VTune should be able to handle such data rate without big overhead. So judging from this result I’m for making itt_task_level_high as default. "